### PR TITLE
JSONAPI

### PR DESCRIPTION
**Pull request by Unnumbered**
private api in jsonapi format

### CODE REVIEW

**Comment by Fivell** [comment]
@Unnumbered , @dmitry-sinina  jsonapi tells us not to use foreign keys as fields but use relationships instead, if we merge this it will be just look like json-api format but not really jsonapi.

**Comment by max-konin** [comment]
@Fivell Hi. I agree with you. But it will take a long time... Because We shall have to  implement API for all model.... 
Mb remove foreign keys from attributes temporarily?

**Comment by Fivell** [comment]
@max-konin in this case how do you see PATCH/POST  let's say for GatewayGroup?

**Comment by Fivell** [comment]
@max-konin  I think there are 2 options 
  - implement real json-api and all take all the benefits from it.
  - do not implement json-api for private API at all.


I don't see how removing foreign keys from API helps because we do need link related objects after all.

**Comment by max-konin** [comment]
@Fivell  what is problem with GatewayGroup?

**Comment by max-konin** [comment]
> I don't see how removing foreign keys from API helps because we do need link related objects after all.

Then, I think we should change status to WIP and start to implement endpoint for all entities

**Comment by max-konin** [comment]
@dmitry-sinina @Fivell Could you help us with error on CI. It mb related with new db's structure....

When we create some model we get error:
```
 ActiveRecord::StatementInvalid:
       PG::NotNullViolation: ERROR:  null value in column "id" violates not-null constraint
       DETAIL:  Failing row contains (null, destination_rate_policy1).
       : INSERT INTO "destination_rate_policy" ("name") VALUES ($1) RETURNING "id"
``` 

ID should be autoincremented. But I can not find it in structure.sql ....

**Comment by dmitry-sinina** [comment]
@max-konin, destination_rate_policy table contains constats. It can't be modified via API or AA.

**Comment by dmitry-sinina** [comment]
Same situation with:
- diversion-policies
- dump-levels
- gateway-rel100-modes
- transport-protocols
- filter-types
- rate-profit-control-modes
- sdp-c-locations
- session-refresh-methods
- dtmf-receive-modes
- dtmf-send-modes
- sensor-levels

**Comment by Fivell** [comment]
@max-konin  just do not crate such records , try to select them from database instead

**Comment by Fivell** [comment]
@max-konin , right

File: Gemfile [hunk 1]
```diff
@@ -32,7 +32,8 @@ gem 'active_admin_sidebar', github: 'activeadmin-plugins/active_admin_sidebar'
 gem 'excelinator', github: 'livingsocial/excelinator'
 
 # REST API
-gem 'active_model_serializers', '~> 0.9.7'
+# gem 'active_model_serializers', '~> 0.9.7'
```
Comment (max-konin) [inline Gemfile]: remove it

**Comment by dmitry-sinina** [comment]
Timezones should be read-only.

**Pull request merged**